index.js

```diff
@@ -78,6 +82,53 @@ class TrezorKeyring extends EventEmitter {
     return this.model;
   }
 
+  /**
+   * Gets the vendor, if known.
+   *
+   * @returns {"trezor" | "onekey"}
+   */
+  getVendor() {
+    return this.vendor;
+  }
+
+  /**
+   * set the vendor name of the hardware wallet
+   * @param {*} features
+   * @returns {'onekey' | 'trezor'}
+   */
+  __setVendor(features) {
+    // If the value of features is null, set vendor to the default value
+    if (features === null) {
+      this.vendor = undefined;
+      return;
+    }
+
+    // No special field, default is trezor device
+    if (
+      !features ||
+      typeof features !== 'object' ||
+      !features.minor_version ||
+      !features.patch_version
+    ) {
+      this.vendor = 'trezor';
+      return;
+    }
+
+    const minorVersion = Number(features.minor_version);
```

Comment:
Since this is just used for equality check there is no need to force it to be a number? Could a hypothetical other device have non-number values here? In that case these intermediates are superfluous.